drivers/sensor/i3g4250d/i3g4250d.c

```diff
@@ -0,0 +1,230 @@
+{
+	val->val1 = (int16_t)(raw_value * RAW_TO_MICRODEGREEPERSEC / 100000LL);
+	val->val2 = (int16_t)(raw_value * RAW_TO_MICRODEGREEPERSEC) % 1000000LL;
+}
+
+static void i3g4250d_channel_convert(enum sensor_channel chan,
+					 uint16_t *raw_xyz,
+					 struct sensor_value *val)
+{
+	uint8_t ofs_start, ofs_stop;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+		ofs_start = ofs_stop = 0U;
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		ofs_start = ofs_stop = 1U;
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		ofs_start = ofs_stop = 2U;
+		break;
+	default:
+		ofs_start = 0U;
+		ofs_stop = 2U;
+		break;
+	}
+
+	for (int i = ofs_start; i <= ofs_stop; i++) {
+		i3g4250d_convert(val++, raw_xyz[i]);
+	}
+}
+
+static int i3g4250d_channel_get(const struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+
+	struct i3g4250d_data *i3g4250d = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+		__fallthrough;
+	case SENSOR_CHAN_GYRO_Y:
+		__fallthrough;
+	case SENSOR_CHAN_GYRO_Z:
+		__fallthrough;
+	case SENSOR_CHAN_GYRO_XYZ:
+		i3g4250d_channel_convert(chan, i3g4250d->angular_rate, val);
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static i3g4250d_dr_t gyr_odr_to_reg(struct sensor_value *val)
+{
+	double odr = sensor_value_to_double(val);
+	i3g4250d_dr_t reg = I3G4250D_ODR_OFF;
+
+	if ((odr > 0.0) && (odr < 100.0)) {
+		reg = I3G4250D_ODR_SLEEP;
+	} else if ((odr >= 100.0) && (odr < 200.0)) {
+		reg = I3G4250D_ODR_100Hz;
+	} else if ((odr >= 200.0) && (odr < 400.0)) {
+		reg = I3G4250D_ODR_200Hz;
+	} else if ((odr >= 400.0) && (odr < 800.0)) {
+		reg = I3G4250D_ODR_400Hz;
+	} else if (odr >= 800.0) {
+		reg = I3G4250D_ODR_800Hz;
+	}
+
+	return reg;
+}
+
+static int i3g4250d_config_gyro(const struct device *dev,
+				enum sensor_attribute attr,
+				const struct sensor_value *val)
+{
+	struct i3g4250d_data *i3g4250d = dev->data;
+	i3g4250d_dr_t dr_reg;
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		dr_reg = gyr_odr_to_reg((struct sensor_value *)val);
+		return i3g4250d_data_rate_set(i3g4250d->ctx, dr_reg);
+	default:
+		LOG_ERR("Gyro attribute not supported");
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int i3g4250d_attr_set(const struct device *dev, enum sensor_channel chan,
+				 enum sensor_attribute attr,
+				 const struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_XYZ:
+		return i3g4250d_config_gyro(dev, attr, val);
+	default:
+		LOG_ERR("attr_set() not supported on this channel %d.", chan);
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static const struct sensor_driver_api i3g4250d_driver_api = {
+	.attr_set = i3g4250d_attr_set,
+	.sample_fetch = i3g4250d_sample_fetch,
+	.channel_get = i3g4250d_channel_get,
+};
+
+static int i3g4250d_init(const struct device *dev)
+{
+	struct i3g4250d_data *i3g4250d = dev->data;
+	uint8_t wai;
+
+	int ret = i3g4250d_spi_init(dev);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = i3g4250d_device_id_get(i3g4250d->ctx, &wai);
+
+	if (ret != 0) {
```

Comment:
there are a few places where a blank line is added between call and error checking, better be consistent

Reply:
I think style checker complained about missing blank lines in case `ret` was declared in this line. Therefore I moved it away in some places. Splitted declaration up and removed the blank lines.